rails/app/controllers/api/v1/external_activities_controller.rb

```diff
@@ -95,4 +95,24 @@ def update_basic
       error("Unable to save external activity options")
     end
   end
+
+  def update
+    begin
+      user, role = check_for_auth_token(params)
+    rescue StandardError => e
+      return error(e.message, 403)
+    end
+
+    external_activity = ExternalActivity.find(params[:id])
+    authorize external_activity
+
+    permitted_params = params.permit(:popup)
```

Comment:
This should be expanded to support updating most of the fields of external_activity.
It should be based on the role of the user.

I'm not sure though if you want to get into this level of detail since it is straying from the initial goal of just adding an API to support the tests. Here are a few options:
- I could make this change myself
- We could make it together
- I could tell you have to make it

For reference the `role` on line 101 shouldn't be used for figuring out the current users permissions in this case. It just indicates if they are a learner or teacher. But we need to know if they are a admin, project admin, or author. And if they are a project admin we need to know if they have permission to edit this external_activity.  The logic or which fields different uses can edit is based on `MaterialSharedPolicy` and several partials that render the external activity edit form:
- rails/app/views/external_activities/_form.html.haml
And its many partials. Here is an incomplete list:
- rails/app/views/shared/_grade_levels_edit.html.haml
- rails/app/views/shared/_subject_areas_edit.html.haml
- rails/app/views/shared/_publication_status_edit.haml
- rails/app/views/shared/_cohorts_edit.html.haml
-